controller/sync.go

```diff
@@ -247,16 +247,18 @@ func (sc *syncContext) sync() {
 		}
 	}
 
-	// any running tasks, lets wait...
-	if tasks.Find(func(t *syncTask) bool { return t.running() }) != nil {
-		sc.setOperationPhase(v1alpha1.OperationRunning, "one or more tasks are running")
-		return
-	}
+	// this block of logic is only for hook/wave syncs - it does not apply retro syncs
+	if tasks.Any(func(t *syncTask) bool { return t.isHook() || t.wave() != 0 }) {
```

Comment:
I think we should skip checks for the last wave (given there are no hooks), not just sync with a single wave.

I think it can be done if you change check as follows:
```
if tasks.Any(func(t *syncTask) bool { return t.isHook() || t.wave() > wave }) {
```